modules/machinery/kvm.py

```diff
@@ -3,10 +3,28 @@
 # This file is part of Cuckoo Sandbox - http://www.cuckoosandbox.org
 # See the file 'docs/LICENSE' for copying permission.
 
+import xml.etree.ElementTree as ET
+
 from lib.cuckoo.common.abstracts import LibVirtMachinery
 
+
 class KVM(LibVirtMachinery):
     """Virtualization layer for KVM based on python-libvirt."""
 
     # Set KVM connection string.
     dsn = "qemu:///system"
+
+    def _get_interface(self, label):
+        xml = ET.fromstring(self._lookup(label).XMLDesc())
+        elem = xml.find("./devices/interface[@type='network']")
+        if elem is None:
+            return elem
+        elem = elem.find("target")
+        if elem is None:
+            return None
```

Comment:
I wrote that function the be a generic NIC finder, for use with Cuckoo and beyond. That case should only occur if someone made a VM without any NICs, in which case they have bigger problems :)